### DISCUSSION TITLE
Let's Encrypt with SEB Server docker container

### DISCUSSION THREAD

**Discussion by blanderson22** [SafeExamBrowser/seb-server#66] [Q&A] [answered]
Has anyone successfully setup Let's Encrypt to issue a cert to the SEB Server docker container? 

Any guide you can point me to?

**Comment by voltere5** [comment, marked as answer]
It is higly adviced to use a dedicated network device like a loadbalancer or firewall which terminates the SSL- traffic and routes the traffic to your backend (seb-guiservice and seb-webservice). for that you will need two different URLs.

if you dont have such an infrastructure you could try to implement something like nginx-proxy-manager or traefik (which both offers easy letsencrypt certificates) and expose these containers to your users.

**Reply by blanderson22** [reply]
Thank you for your reply. So, for instance if I deploy traefik do I then replace the nginx container that is included in the SEB Server dockerhub with traefik?

**Reply by anhefti** [reply]
Hi, yes this should be possible but we have never tried that with traefik so far. You have to setup and test this by yourself but this should work as well.

In general SEB Server is just a Java enterprise server within a docker container that can be setup all-in-one (like in dockerhub) or in a distributed setup for scaling (kubernetes). In the distributed setup the SEB Server can be split into two parts.

In the all-in-one setup, SEB Server exposes internal docker port (http) to 8080 or whatever you have configured. The docker-compose creates a  docker-network for internal communication and exposes its ports to the host-system within the nginx reverse proxy that also do the TLS termination. So you can either replace the nginx with a corresponding and convenient traefik docker definition in the docker-compose file or you can remove it entirely from the given docker-compose and expose the http Port of the SEB Server container directly to your internal network where another/existing reverse-proxy is doing the TLS termination. This is completely up to you or depends on your needs and the infrastructure you have.

**Reply by blanderson22** [reply]
I think I'm close to having this sorted out. Is it possible that I could pay you for a couple hours of your time to take a look at my server? 

I have Nginx Proxy Manager running in a docker container and I've removed the seb-server-proxy from the SEB docker-compose file. I've setup a reverse proxy definition in the Nginx Proxy Manager for seb-server, but alas, the specified domain isn't loading the SEB gui.

**Reply by blanderson22** [reply]
Woohoo! I have the SEB interface loading now, but I'd still love to have you review the environment if you're willing ... I'll pay you :-)

**Reply by anhefti** [reply]
Hi and congratulations :-) 

Unfortunately direct payment is not possible. For review it would be possible to post your configuration files here so we can have a look at it, if you wish. 

In the case you want to use SEB Server in production within your institution and you want to have guaranteed support, please considering a [SEB Alliance](https://www.safeexambrowser.org/alliance/members.html)  membership.

And also don't hesitate to reach out here when you have other questions/problems about SEB Server. We always try to answer/solve it if we can and have time to do so.

**Reply by blanderson22** [reply]
Thank you. I will inquire internally about the possibility of becoming an Alliance member.
